src/GraphQL.Tests/Execution/VariablesTests.cs

```diff
@@ -185,9 +185,8 @@ public void fail_on_incorrect_value()
               fieldWithObjectInput(input: ["foo", "bar", "baz"])
             }
             """;
-        var expected = "{ \"fieldWithObjectInput\": null }";
 
-        var result = AssertQueryWithErrors(query, expected, rules: Enumerable.Empty<IValidationRule>(), expectedErrorCount: 1, executed: true);
+        var result = AssertQueryWithErrors(query, null, rules: Enumerable.Empty<IValidationRule>(), expectedErrorCount: 1, executed: false);
         result.Errors[0].Message.ShouldBe("Error trying to resolve field 'fieldWithObjectInput'.");
         result.Errors[0].InnerException.Message.ShouldStartWith("Expected object value for 'TestInputObject', found not an object '[\"foo\", \"bar\", \"baz\"]'.");
```

Comment:
Tests like this we can see:
- Request is not executed if the argument literal cannot parse correctly
- Errors can now be improved upon (either in this PR or a follow-up PR) as they do not occur within a field resolver but rather during document validation